dev/PushNotifications/PushNotifications.idl

```diff
@@ -0,0 +1,134 @@
+    CompletedSuccess, // The request completed successfully
+    CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+};
+
+// The PushNotificationChannel Progress result
+struct PushNotificationCreateChannelStatus
+{
+    // The last extended error we failed Channel requests on that caused the inprogress retry status. S_OK if this is just progress status.
+    HRESULT extendedError;
+
+    // Either InProgress or InProgressRetry status
+    PushNotificationChannelStatus status;
+
+    // Total Retries so far
+    UInt32 retryCount;
+};
+
+runtimeclass PushNotificationChannel
+{
+    PushNotificationChannel(Windows.Networking.PushNotifications.PushNotificationChannel channel);
+
+// The Channel Uri for app to Post a notification to.
+Windows.Foundation.Uri Uri { get; };
+
+// Expiration of the Channel
+Windows.Foundation.DateTime ExpirationTime { get; };
+
+// Unsubscribes the channel
+void Close();
+
+// In-memory Event handler for Push Notifications
+event Windows.Foundation.TypedEventHandler<PushNotificationChannel, PushNotificationReceivedEventArgs> PushReceived;
+}
+
+runtimeclass PushNotificationCreateChannelResult
+{
+    PushNotificationCreateChannelResult(
+        PushNotificationChannel channel,
+        HRESULT extendedError,
+        PushNotificationChannelStatus status);
+
+// The Push channel associated with the Result. Null if InProgress or completion failed
+PushNotificationChannel Channel { get; };
+
+// More detailed error code in addition to the ChannelStatus state.
+HRESULT ExtendedError{ get; };
+
+// The Status of the ChannelComplete operation
+PushNotificationChannelStatus Status { get; };
+};
+
+runtimeclass PushNotificationRegistrationToken
+{
+    PushNotificationRegistrationToken(
+        UInt64 cookie,
+        Windows.ApplicationModel.Background.BackgroundTaskRegistration taskRegistration);
+
+// The cookie from CoRegisterClassObject
+UInt64 Cookie{ get; };
+
+// The Registration token for the Push Trigger
+Windows.ApplicationModel.Background.BackgroundTaskRegistration TaskRegistration { get; };
+};
+
+static runtimeclass PushNotificationManager
+{
+    // Register an activator using an ActivationInfo context and return a RegistrationToken
+    static PushNotificationRegistrationToken RegisterActivator(PushNotificationActivationInfo details);
+
+// Unregister any activator if present using a token and registrationKind
+// 1) If kind = PushTrigger is specified, the trigger itself will be removed
+// 2) If kind = ComActivator is specified, the Project Reunion Background Task component will no longer act as an InProc COM Server
+static void UnregisterActivator(PushNotificationRegistrationToken token, PushNotificationRegistrationKind kind);
+
+// Request a Push Channel with an encoded RemoteId from WNS. RemoteId is an AAD identifier GUID
```

Comment:
Comment is confusing. First sentence says that it's a RemoteId from WNS. Second sentence says that it is an AAD identifier. So which is it?

Reply:
Raymond, RemoteId is nothing but AAD identified. I think the first statement "Request a Push Channel with an encoded RemoteId from WNS" - this looks ok to me. But I will modify the statement to keep it simple. Thanks.

Reply:
@sharath2727 Comment doesn't seem to offer any value; I'd delete it if possible. Or if that's not possible, delete the second sentence. _AAD identifier GUID_ isn't a real thing in Azure and sounds funny when you expand the acronyms: "Azure Active Directory identifier globally unique identifier"

Reply:
If "RemoteId from WNS" and "AAD identifier GUID" are the same thing, then saying the same thing in two different ways creates confusion. It's like writing "Pass a BCP 47 language tag. It should conform to RFC 5646." It sounds like the parameter needs to satisfy two different requirements, but what you don't know is that BCP 47 and RFC 5646 are just two names for the same thing.

At least remove the second part if it is redundant. If the entire comment doesn't add value, then remove it entirely.



Remove the redundant and confusing sentence, possibly even the whole sentence if it adds no value.

Reply:
I will simplify this. Thanks.